### PR TITLE
test(general): ignore dir not found error during delete action in robustness tests

### PR DESCRIPTION
**Pull request by chaitalisg**
The robustness tests perform delete actions on random subdirectories and files. If the actions encounter `directory not found` error, the framework throws `no-op` error. This PR changes the test to ignore the `no-op` errors thrown in such scenario. Making this change will reduce the failure frequency of the test when delete actions on the source directory result in a no-op.

### CODE REVIEW

File: tests/robustness/multiclient_test/multiclient_test.go [hunk 1]
```diff
@@ -52,10 +52,16 @@ func TestManySmallFiles(t *testing.T) {
 		require.NoError(t, err)
 
 		_, err = eng.ExecAction(ctx, engine.DeleteRandomSubdirectoryActionKey, deleteDirOpts)
-		require.NoError(t, err)
+		// ignore the dir-not-found errors
+		if err != robustness.ErrNoOp {
+			require.NoError(t, err)
```
Comment (julio-lopez) [inline tests/robustness/multiclient_test/multiclient_test.go:57]: Please address the linter errors.

File: tests/robustness/multiclient_test/multiclient_test.go [hunk 2]
```diff
@@ -52,10 +52,16 @@ func TestManySmallFiles(t *testing.T) {
 		require.NoError(t, err)
 
 		_, err = eng.ExecAction(ctx, engine.DeleteRandomSubdirectoryActionKey, deleteDirOpts)
-		require.NoError(t, err)
+		// ignore the dir-not-found errors
```
Comment (julio-lopez) [inline tests/robustness/multiclient_test/multiclient_test.go:55]: Create a helper for executing and ignoring the no-op error for delete-type actions, and then call that helper here and below.

File: tests/robustness/multiclient_test/multiclient_test.go [hunk 3]
```diff
@@ -186,7 +198,10 @@ func TestRandomizedSmall(t *testing.T) {
 		//nolint:forbidigo
 		for st.Elapsed() <= *randomizedTestDur {
 			err := tryRandomAction(ctx, t, opts)
-			require.NoError(t, err)
+			// ignore the dir-not-found errors
+			if err != robustness.ErrNoOp {
+				require.NoError(t, err)
+			}
```
Comment (julio-lopez) [inline tests/robustness/multiclient_test/multiclient_test.go:204]: is this the right check to perform here?
what type of actions does it apply to?

**Review by julio-lopez — CHANGES_REQUESTED** [review]

File: tests/robustness/multiclient_test/multiclient_test.go (hunk 2 above)
Comment (chaitalisg) [reply tests/robustness/multiclient_test/multiclient_test.go]: done

File: tests/robustness/multiclient_test/multiclient_test.go (hunk 1 above)
Comment (chaitalisg) [reply tests/robustness/multiclient_test/multiclient_test.go]: done

File: tests/robustness/multiclient_test/multiclient_test.go (hunk 3 above)
Comment (chaitalisg) [reply tests/robustness/multiclient_test/multiclient_test.go]: Good catch. The function `tryRandomAction` already masks the no-op errors, so no changes are needed here.

File: tests/robustness/multiclient_test/multiclient_test.go [hunk 4]
```diff
@@ -215,3 +215,15 @@ func tryRandomAction(ctx context.Context, t *testing.T, opts engine.ActionOpts)
...
+// tryExecAction runs the given engine action with options and masks no-op errors.
+func tryExecAction(ctx context.Context, t *testing.T, action engine.ActionKey, actionOpts map[string]string) error { //nolint:thelper
+	_, err := eng.ExecAction(ctx, action, actionOpts)
+	// Ignore the dir-not-found error, wrapped as no-op error.
+	if errors.Is(err, robustness.ErrNoOp) {
+		t.Log("Delete action resulted in no-op")
+		return nil
+	}
+
+	return err
```
Comment (julio-lopez) [inline tests/robustness/multiclient_test/multiclient_test.go:228]: What about moving the `require.NoError` check to the helper and removing the returned `error` from the function signature?

Note: it is OK to use "testing" patterns in the robustness framework.

**Review by julio-lopez — COMMENTED** [review]
Moving in the right direction. See inline comments

File: tests/robustness/multiclient_test/multiclient_test.go [hunk 5]
```diff
@@ -215,3 +215,15 @@ func tryRandomAction(ctx context.Context, t *testing.T, opts engine.ActionOpts)
 
 	return err
 }
+
+// tryExecAction runs the given engine action with options and masks no-op errors.
+func tryExecAction(ctx context.Context, t *testing.T, action engine.ActionKey, actionOpts map[string]string) error { //nolint:thelper
```
[the remark this answers is not in the document]
Comment (julio-lopez) [reply tests/robustness/multiclient_test/multiclient_test.go:220]: Remove the `//nolint.thelper` annotation.
Instead, call `t.Helper()` at the start of the helper.

[the remark this answers is not in the document]
Comment (chaitalisg) [reply tests/robustness/multiclient_test/multiclient_test.go]: Added the validation for the action type. Added `t.Helper()`

File: tests/robustness/multiclient_test/multiclient_test.go (hunk 4 above)
Comment (chaitalisg) [reply tests/robustness/multiclient_test/multiclient_test.go]: done

**Review by julio-lopez — APPROVED** [review]

**Pull request merged**